client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -150,19 +170,42 @@ const CheckoutThankYou = React.createClass( {
 	},
 
 	render() {
-		const classes = classNames( 'checkout-thank-you', {
-			'is-placeholder': ! this.isDataLoaded()
-		} );
-
 		let purchases = null,
-			wasJetpackPlanPurchased = false;
+			wasJetpackPlanPurchased = false,
+			wasOnlyDotcomPlanPurchased = false;
 		if ( this.isDataLoaded() && ! this.isGenericReceipt() ) {
 			purchases = getPurchases( this.props );
 			wasJetpackPlanPurchased = purchases.some( isJetpackPlan );
+			wasOnlyDotcomPlanPurchased = purchases.every( isPlan );
+		}
+
+		const userCreatedMoment = moment( this.props.user.date );
+		const isNewUser = userCreatedMoment.isAfter( moment().subtract( 2, 'hours' ) );
+		const isPaidNuxStreamlinedAbTest = abtest( 'paidNuxStreamlined' ) === 'streamlined';
+
+		// this placeholder is using just wp logo here because two possible states do not share a common layout
+		if ( ! purchases && ! this.isGenericReceipt() ) {
+			// disabled because we use global loader icon
+			/* eslint-disable wpcalypso/jsx-classname-namespace */
+			return (
+				<div className="wpcom-site__logo noticon noticon-wordpress"></div>
+			);
+			/* eslint-enable wpcalypso/jsx-classname-namespace */
+		}
+
+		// streamlined paid NUX thanks page
+		if ( isPaidNuxStreamlinedAbTest && isNewUser && wasOnlyDotcomPlanPurchased ) {
+			return (
+				<Main className="checkout-thank-you">
+					<PlanThankYouCard siteId={ this.props.selectedSite.ID } />
```

Comment:
What if, instead of relying on passing down `siteId`, we get the `siteId` with `getSelectedSiteId` in `PlanThankYouCard` `mapStateToProps` function? We can still pass down `siteId`, but if we don't, just use the siteId from `getSelectedSiteId`.

I think this would make the component less requiring of any props. I also don't like how `this.props.selectedSite` gets passed down to `CheckoutThankYouComponent`. It originates from `sites.getSelectedSite()`.